### ISSUE TITLE
Type variable across types gives an error when using Generics

### ISSUE THREAD

**Issue by flisboac** [python/mypy#11854]
**Bug Report**

When using a TypeVar defined in another local file, to parameterize a Generic, the following error is given:

> Free type variable expected in Generic[...]

**To Reproduce**

Create file `value.py`:

```py
from typing import TypeVar

TValue = TypeVar('TValue', covariant=True)
```

Create `service.py`:

```py
from abc import ABC
from typing import Generic

from  .value import TValue

class Service(ABC, Generic[TValue]):  # Error occurs here, on `Generic`
    ...
```

**Expected Behavior**

Should not give any errors (I think).

**Actual Behavior**

An error is produced, when parameterizing  `Generic`:

> Free type variable expected in Generic[...]

**Your Environment**

- Mypy version used: 0.930
- Mypy command-line flags: (none)
- Mypy configuration options from `mypy.ini` (and other config files): (none was created, at least explicitly)
- Python version used: CPython 3.8.10
- Operating system and version: Ubuntu 20.04 LTS on WSL2

**Comment by A5rocks** [comment]
I cannot reproduce this, though I had to make the following changes to make this run (maybe you left something out?):
 - add `Generic` to the imports for `value.py`
 - add an (empty) `__init__.py`

**Comment by flisboac** [comment]
The `IValueProcessor` was a last-minute addition. I forgot to also import `Generic` in `value.py` in the example, but on my source code it is imported, and no error is emitted.

And indeed, `__init__.py` was missing. Never thought it would be a requirement. Mypy should be able to detect TypeVar imports the same way it detects and understands other normal imports (non-typing elements). Isn't this a bug?

**Comment by nstarman** [comment]
I also get this error with mypy 0.991 on VSCode.

**Comment by ikonst** [comment]
This stems from `--ignore-missing-imports` being the default in some configuration, which causes
```python
from not_found_module import TValue
```
not to be flagged, but instead `TValue` being `Any`.

The issue here is that "Free type variable expected in Generic[...]" error message is not clear enough.

**Comment by pranavrajpal** [comment]
I believe this issue seems to occur with vscode-python for everyone here because vscode-python changes the default mypy arguments to include `--ignore-missing-imports` (see [here](https://github.com/microsoft/vscode-python/blob/d844f7d2c8951b541d371ecafe0663e3b1a0490f/package.json#L678-L684)).

For any of you running into this issue, try seeing if you can reproduce this by running mypy from the command line since I suspect you'd get a more informative error that way.

Also, if you want to use mypy from VS Code I'd personally recommend [this extension](https://marketplace.visualstudio.com/items?itemName=matangover.mypy), since from my experience I don't think it changes the default mypy arguments and it generally matches the results from the command line version more accurately.

**Comment by ikonst** [comment]
`--ignore-missing-imports` is pretty common. The pre-commit hook also adds it.
